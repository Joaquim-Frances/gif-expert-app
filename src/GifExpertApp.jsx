import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


const GifExpertApp = ({ defaultCategories = [] }) => {

   
    const [categories, setCategories] = useState( defaultCategories );


    return (
        <div className="animate__animated animate__fadeInRightBig">
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>
            <ol>
                { 
                    categories.map( category => {
                        return(
                            <GifGrid 
                                key = { category }
                                category={ category }
                        />)
                    })
                }
            </ol>
        </div>

    )
};

export default GifExpertApp;