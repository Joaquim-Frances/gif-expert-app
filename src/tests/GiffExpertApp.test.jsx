const { shallow } = require("enzyme");
import React from 'react';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';



describe('Testing GiffExpertApp main app', () => {

    test('should match the snapshot correctly', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
        
    })

    test('should show a list of categories', () => {

        const categories = ['Son Goku', 'Dr. Slump'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect(wrapper).toMatchSnapshot;
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
    
})
