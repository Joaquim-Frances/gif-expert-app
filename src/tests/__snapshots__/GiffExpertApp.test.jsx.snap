// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GiffExpertApp main app should match the snapshot correctly 1`] = `
<div
  className="animate__animated animate__fadeInRightBig"
>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
